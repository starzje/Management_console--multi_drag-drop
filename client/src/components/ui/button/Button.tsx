import * as React from "react";

interface ButtonProps {
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  text: string;
  btnClass: string;
  isDisabled?: boolean;
  toggleGameType?: number;
}

export const Button: React.FC<ButtonProps> = ({ btnClass, onClick, isDisabled, text }) => {
  return (
    <button disabled={isDisabled} onClick={onClick} className={btnClass}>
      {text}
    </button>
  );
};
