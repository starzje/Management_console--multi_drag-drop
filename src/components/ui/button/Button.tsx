import * as React from "react";

interface ButtonProps {
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  text: string;
  btnClass: string
  isDisabled?: boolean;
  toggleGameType?: number;
}

const Button: React.FC<ButtonProps> = ({
  btnClass,
  onClick,
  isDisabled,
  text,
}) => {
  return (
    <button
      disabled={isDisabled}
      onClick={onClick}
      className={btnClass}
    >
      {text}
    </button>
  );
};

export default Button;
