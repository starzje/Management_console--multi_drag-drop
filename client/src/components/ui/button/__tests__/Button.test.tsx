import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Button } from "../Button";
import "@testing-library/jest-dom/extend-expect";

describe("Button", () => {
  it("should render with text and class", () => {
    const { getByText, container } = render(<Button btnClass="my-btn-class" text="Click me" onClick={() => {}} />);

    const button = getByText("Click me");
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass("my-btn-class");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the button if `isDisabled` prop is passed", () => {
    const { getByText } = render(
      <Button btnClass="my-btn-class" text="Click me" onClick={() => {}} isDisabled={true} />,
    );

    const button = getByText("Click me");
    expect(button).toBeDisabled();
  });

  it("should call the `onClick` prop when clicked", () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button btnClass="my-btn-class" text="Click me" onClick={handleClick} />);

    const button = getByText("Click me");
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
