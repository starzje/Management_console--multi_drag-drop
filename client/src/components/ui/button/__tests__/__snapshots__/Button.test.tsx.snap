// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with text and class 1`] = `
<button
  class="my-btn-class"
>
  Click me
</button>
`;
